nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkCommand.java

```diff
@@ -74,17 +80,36 @@ public String getQuery() {
         return query;
     }
 
-    public BulkCommand withOperation(String operation) {
-        this.operation = operation;
+    public BulkCommand withAction(String action) {
+        this.action = action;
+        return this;
+    }
+
+    public String getAction() {
+        return action;
+    }
+
+    public BulkCommand withParams(Map<String, Serializable> params) {
```

Comment:
This will cause pb on avro conversion, serializable is a java codec non interroperable, 
also if these params are going to be contributed with XML it might be better to start with a map of String

Reply:
Converted to Map<String, String>.

Reply:
+1 for Map<String, String>.  I have been using this approach for the various descriptor configurations with an interface. https://github.com/nuxeo/ai-core/blob/master/nuxeo-pipes-core/src/main/java/org/nuxeo/runtime/stream/pipes/streams/Initializable.java